orchagent/portsorch.cpp

```diff
@@ -2304,27 +2308,21 @@ void PortsOrch::doPortTask(Consumer &consumer)
             auto port_id = m_portList[alias].m_port_id;
             auto hif_id = m_portList[alias].m_hif_id;
 
-            if (m_portList[alias].has_dependency())
-            {
-                // Port has one or more dependencies, cannot remove
-                SWSS_LOG_WARN("Please remove port dependenc(y/ies):%s",
-                               m_portList[alias].print_dependency().c_str());
-                it++;
-                continue;
-            }
-
-            deinitport(alias, port_id);
-
-            SWSS_LOG_NOTICE("Removing hostif %lx for Port %s", hif_id, alias.c_str());
-            sai_status_t status = sai_hostif_api->remove_hostif(hif_id);
-            if (status != SAI_STATUS_SUCCESS)
-            {
-                throw runtime_error("Remove hostif for the port failed");
+            if (m_portList[alias].m_init) {
+                deinitport(alias, port_id);
+                SWSS_LOG_NOTICE("Removing hostif %lx for Port %s", hif_id, alias.c_str());
+                sai_status_t status = sai_hostif_api->remove_hostif(hif_id);
+                if (status != SAI_STATUS_SUCCESS)
+                {
+                    throw runtime_error("Remove hostif for the port failed");
+                }
+                m_portList[alias].m_init = false;
             }
 
             if (!removePort(port_id))
             {
-                throw runtime_error("Delete port failed");
+                it++;
```

Comment:
code below removePort() could be called many times.  Do you see issues?
```
    Port p;
    if (getPort(port_id, p))
    {
        PortUpdate update = {p, false };
        notify(SUBJECT_TYPE_PORT_CHANGE, static_cast<void *>(&update));
    }
```

Reply:
No

Reply:
To make it symmetric with addport() and void any re-entering issue for notify() for different observers. I would suggest to move this part of code to outside, and we should not call it again and again during the retry.

Reply:
Agree. Moved it out of the function and inside the init block.